slack-server-integration-common/src/test/java/com/atlassian/plugins/slack/api/client/DefaultSlackClientWithMockedServerTest.java

```diff
@@ -0,0 +1,214 @@
+package com.atlassian.plugins.slack.api.client;
```

Comment:
New test spins a mock HTTP server and instantiates a client so that it actually runs with all interceptors. It also allow inspecting the logs generated by tests at a granular level.